add Dirac multicore options and fix disk checker

mesmith75 opened the pull request:
Fixes #72 

I think I managed to find all the Dirac handlers. I didn't update LSST and SNO+ as I don't think they have been updated for python3.

mesmith75 committed:
add Dirac multicore options

mesmith75 committed:
Merge branch 'develop' into multicore

mesmith75 committed:
Merge branch 'develop' into multicore

egede reviewed (commented):
@alexanderrichards Does the (older) Dirac client and server at Imperial accept these options (it might not do anything with them but they should not cause an error).

alexanderrichards commented:
> @alexanderrichards Does the (older) Dirac client and server at Imperial accept these options (it might not do anything with them but they should not cause an error).

I don't know off hand. I'll find out

mesmith75 committed:
Merge branch 'develop' into multicore

mesmith75 commented:
What is the plan for this?

egede commented on ganga/GangaDirac/Lib/RTHandlers/DiracRTHScript.py.template:
```diff
@@ -27,5 +27,7 @@ j.setParametricInputData(###PARAMETRIC_INPUTDATA###)
 
 # submit the job to dirac
 j.setPlatform( '###PLATFORM###' )
+j.setNumberOfProcessors(minNumberOfProcessors=###MIN_PROCESSORS###, maxNumberOfProcessors=###MAX_PROCESSORS###)
+
```
Can we just wrap this in a `try-except` clause and ignore it if the client raises an `AttributeError`

mesmith75 replied:
Ah yes, good idea.

mesmith75 replied:
This catches the `AttributeError` for very old Dirac versions, and a `TypeError` as the keyword arguments changed. In both cases it proceeds silently.

mesmith75 committed:
add try except

mesmith75 committed:
fix

alexanderrichards commented:
> I don't know off hand. I'll find out

Sorry for the delay, I checked and Imperial are running at least v6r22p26 and that version has this method so should be all good. However a try: except: block couldn't hurt

alexanderrichards commented:
> Sorry for the delay, I checked and Imperial are running at least v6r22p26 and that version has this method so should be all good. However a try: except: block couldn't hurt

Ok scratch that, while they do have the method at v6r22p26 it doesn't take min/max as args, instead simply a number so you will need to try/catch

mesmith75 commented:
Yes, that is the TypeError

mesmith75 committed:
Merge branch 'develop' into multicore

mesmith75 committed:
Merge branch 'develop' into multicore

egede committed:
Merge branch 'develop' into multicore

egede reviewed (dismissed):
It would be good to update the docstring as well to include an example, and mention that setting min above 1 might limit places that job can run.

egede commented:
@mesmith75 There is a test failure. Not clear to me why from a quick look at it ...

mesmith75 committed:
fix literal

mesmith75 committed:
fix disk checker for non afs

mesmith75 committed:
update docstring

mesmith75 commented:
I think I have fixed the tests. They were failing because the disk checker now logs an error when something goes wrong.

The disk checker was not working on non-AFS file systems. Looking at the old code and output of `df` I'm not sure this function ever worked but the error was hidden in the debug. Updating to a regex should do it.

Also fixed a deprecation warning in a Condor regex whilst at it.

Edit: spoke too soon. I'm not sure how these tests ever passed. They seem unrelated to the original content of this PR.

mesmith75 committed:
fix disk checker

mesmith75 committed:
fix GListApp

mesmith75 committed:
fix GListApp

mesmith75 committed:
Merge branch 'develop' into multicore
